### PR TITLE
Add LibPSTerra to AlpsLib, implement API-Connection

### PR DESCRIPTION
**Pull request by GrJo**
(**ON HOLD** -  I see the PlotSystemTerra-Plugin has seen some updates for 1.20 from LordTuxn, which are not included in this Pull Request. Please wait while i move these changes here.)

As planned with the BTE BuildTeamsTools developers, this moves most of the code from AlpsLib/PlotSystemTerra into the AlpsLib Library, so that it can be used from plugins with minimal code. The corresponding change to PlotSystemTerra is also ready to be pulled back from this fork to main - i will prepare the pull-request as well. 

I have created a new submodule "LibPSTerra" in AlpsLib for that, and bumped the version numbers to 1.1.0 across the board for this change.

Also, this change finally implements the **Datamode "API"** by adding an appropriate interface `Connection` thats implemented by `DatabaseConnection `(thats where the database query code has moved to) and `NetworkAPIConnection` (the new code thats using the [API](https://buildtheearth.github.io/network-api/). Along the way, i added some classes to represent/group the data thats returned by the database/network queries (`Server.java, Plot.java, Country.java`)

Everything has been tested as far as I could, i also added some unit-tests/integration tests in the package test/java/TestConnections.java

### CODE REVIEW

**Review by LordTuxn — CHANGES_REQUESTED** [review]
lgtm, please implement the requested changes to avoid duplicate code.

File: libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/ItemBuilder.java
[the remark this answers is not in the document]
Comment (GrJo) [reply libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/ItemBuilder.java:1]: the Itembuilder from AlpsLib/utils does not work in 1.12 (AdventureAPI). I moved the Itembuilder from libpsterra there and renamed it LegacyItembuilder.
[12e472d](https://github.com/AlpsBTE/Alps-Lib/pull/1/commits/12e472d182575b37b5a1c8e6ddb0858274cdb50a)
[fe6fc5a](https://github.com/AlpsBTE/Alps-Lib/pull/1/commits/fe6fc5aeeddc8cd6afe2e94fba337276a602a231)

File: libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/LoreBuilder.java
[the remark this answers is not in the document]
Comment (GrJo) [reply libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/LoreBuilder.java:1]: done
[fe6fc5a](https://github.com/AlpsBTE/Alps-Lib/pull/1/commits/fe6fc5aeeddc8cd6afe2e94fba337276a602a231)

File: libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/Updater.java [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+        waitThread();
+        return version;
+    }
+
+    public static void notifyUpdate(String newVersion){
+        for(Player p : Bukkit.getOnlinePlayers())
+            notifyUpdate(p, newVersion);
+
+        Bukkit.getConsoleSender().sendMessage(" ");
+        Bukkit.getConsoleSender().sendMessage("§c[BuildTeam Plugin] §eThe server automatically installed a new update (v" + newVersion + ").");
```
[the remark this answers is not in the document]
Comment (GrJo) [reply libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/Updater.java:122]: I did not touch that code. Thats as-is in the current Plot-System-Terra plugin... 
https://github.com/AlpsBTE/Plot-System-Terra/blob/b365228b887fdb26ea1d1a59994e202d6d9a1605/src/main/java/com/alpsbte/plotsystemterra/utils/Updater.java#L120

File: libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/Utils.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    // Head Database API
+    public static HeadDatabaseAPI headDatabaseAPI;
+
+    /**
+     * Prefix used for all command permissions.
+     */
+    public static final String permissionPrefix = "plotsystem";
+
+    public static ItemStack getItemHead(String headID) {
```
[the remark this answers is not in the document]
Comment (GrJo) [reply libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/Utils.java]: using AlpsLib/utils for Head-Query - done
[a3191e0](https://github.com/AlpsBTE/Alps-Lib/pull/1/commits/a3191e08bd188d8c5286b0697178133ca8a185af)

File: libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/Utils.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    public static String getErrorMessageFormat(String error, FileConfiguration config) {
+        return messagePrefix(config) + config.getString(ConfigPaths.MESSAGE_ERROR_COLOUR) + error;
+    }
+
+    public static boolean hasPermission(CommandSender sender, String permissionNode) {
+        return sender.hasPermission(permissionPrefix + "." + permissionNode);
+    }
+
+    public static Integer tryParseInt(String integer) {
```
[the remark this answers is not in the document]
Comment (GrJo) [reply libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/utils/Utils.java]: well spotted. removed the code duplicate.
[72258b5](https://github.com/AlpsBTE/Alps-Lib/pull/1/commits/72258b5c98e3f13b0d12bcf66923f27766c4979f)

File: libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/core/config/ConfigNotImplementedException.java
[the remark this answers is not in the document]
Comment (GrJo) [reply libpsterra/src/main/java/com/alpsbte/alpslib/libpsterra/core/config/ConfigNotImplementedException.java:1]: removed the duplicated file
[7466df7](https://github.com/AlpsBTE/Alps-Lib/pull/1/commits/7466df7a2712009c51f42e68e3a5e782c5f6aacb)

**Comment by GrJo** [comment]
@LordTuxn: can we skip the rest of the Config-Handling for now? I have removed the Exception-Duplicate. But the rest of the config handling is sufficiently different that i dont want to touch this right now.

**Comment by LordTuxn** [comment]
> @LordTuxn: can we skip the rest of the Config-Handling for now? I have removed the Exception-Duplicate. But the rest of the config handling is sufficiently different that i dont want to touch this right now.

Yeah sure, it's not that important as long as it works.

**Review by LordTuxn — APPROVED** [review]

**Pull request merged**